Travis

stirlingv opened the pull request:
Added travis tests for proxy server, admin terminal, analytics terminal, elastic search, elk, mesos, and gocd

stirlingv committed:
Added travis.yml and supporting docker file.

stirlingv committed:
Added admin terminal to travis tests.

Added admin terminal to travis tests.

stirlingv committed:
Added proxy server to travis tests.

stirlingv committed:
Added elastic_search to travis tests.

stirlingv committed:
Added elk playbook to travis tests.

stirlingv committed:
Added analytics terminal to travis tests.

    pick 324de53 Added elk playbook to travis tests.
    pick 522f051 Added analytics terminal to travis tests.
    pick a254afd Added mesos to travis tests and instructions to run travis locally
    s a8adc7d adding all travis tests sothe will be valid for PR

stirlingv committed:
pick 522f051 Added analytics terminal to travis tests.
    pick a254afd Added mesos to travis tests and instructions to run travis locally

    p 33fa5b7 Added gocd to travis tests.

tanderegg commented on .travis.yml:
```diff
@@ -0,0 +1,147 @@
+    PLAYBOOK_LOCATION=/aurora/deploy/deploy_admin_terminal.yml 
+    EXTRA_VAR_DEF='{"install_glusterfs": false, "install_python": false, "install_logstash": false}'
+
+ # analytics terminal - development_terminal and r_terminal replace analytics_terminal
```
Let's remove this section if it is no longer needed.

tanderegg commented on .travis.yml:
```diff
@@ -0,0 +1,147 @@
+  - python
+
+before_install:
+  #- sudo apt-get update
```
Is there any harm in running this?  If not, let's uncomment it.  If so, let's remove it completely.

tanderegg commented on .travis.yml:
```diff
@@ -0,0 +1,147 @@
+  - >
+    docker exec --tty "$(cat ${container_id})" env TERM=xterm ansible-playbook "${PLAYBOOK_LOCATION}" --extra-vars "@/tmp/extra_vars.json" 
+
+#extra log info
```
Can you preface all these comments with something describing what we are trying to fix here?  It may not be obvious down the road, or to new folks ;)

tanderegg commented on Dockerfile:
```diff
@@ -0,0 +1,20 @@
+# Install Ansible
+RUN yum -y install epel-release
+RUN yum -y install git ansible sudo wget
+# RUN yum clean all
```
Is there any harm in running this?  If not, uncomment, if so, remove.

tanderegg commented on Dockerfile:
```diff
@@ -0,0 +1,20 @@
+
+# Add new user called vagrant
+RUN useradd -ms /bin/bash vagrant
+# USER vagrant
```
Please remove these comments if no longer needed.

tanderegg commented on README.md:
```diff
@@ -33,6 +33,48 @@ server is for and how it is meant to be used.
+1. sudo apt-get update
+1. apt-cache policy docker-engine
+1. sudo apt-get install docker-engine (May neeed --force-yes)
+<!-- 1. sudo dpkg-divert --local --rename --add /sbin/initctl
```
We don't need this section anymore, right?

tanderegg commented on deploy/deploy_analytics_terminal.yml:
```diff
@@ -1,100 +1,25 @@
 ---
-- hosts: analytics_terminal
+- hosts: analytics_terminals
```
Shouldn't this be `analytics_terminal`?  It should match the group name in the Travis file, and the vagrant_hosts file.

tanderegg commented on deploy/deploy_analytics_terminal.yml:
```diff
@@ -1,100 +1,25 @@
-      with_items: python3_packages
-
-- hosts: analytics_terminal
+- hosts: analytics_terminals
```
Same here

tanderegg commented on deploy/deploy_development_terminal.yml:
```diff
@@ -0,0 +1,77 @@
+- hosts: analytics_terminal
+  become: yes
+  tasks:
+    - name: Install packages development packages
```
While you're at it, can you fix this typo?  Thanks!

tanderegg commented on deploy/roles/eod/defaults/main.yaml:
```diff
@@ -2,6 +2,8 @@
 #==================#
 ### EoD SETTINGS ###
 #==================#
+install_desktop: true
```
Actually, the default here should be "False", since we do not install EOD when running on Vagrant.

tanderegg commented on deploy/roles/gocd_server/tasks/main.yml:
```diff
@@ -18,6 +18,13 @@
     get_md5: False
   register: gocd_rpm
 
+# - debug: 
```
Can you remove these comments, thanks!

stirlingv committed:
# This is a combination of 3 commits.

    p aa8e4b0 Uncommenting all travis configurations to run tests for all playbooks

stirlingv committed:
uncommenting all travis configurations to run tests for all playbooks

    s 9ea6c72 adding 0 num parameter
    s 3d4856b number of processes (forks) must be >=1

tanderegg commented on deploy/roles/eod/tasks/main.yaml:
```diff
@@ -315,4 +317,4 @@
   when: custom_repo
   tags:
-    - eod
+    - eoded mesos to travis tests and instructions to run travis locally
```
@stirlingv Seems that a comment ended up in this list, it should just be "eod"

stirlingv committed:
fixing typos based on PR review
